posthog/hogql/base.py

```diff
@@ -34,6 +34,20 @@ def accept(self, visitor):
             return visitor.visit_unknown(self)
         raise NotImplementedError(f"{visitor.__class__.__name__} has no method {method_name}")
 
+    def __str__(self):
+        from posthog.hogql.printer import print_prepared_ast
+        from posthog.hogql.context import HogQLContext
+
+        printed = print_prepared_ast(
```

Comment:
what happens if there's an issue with the printer and an exception is thrown?

would it make sense to print the exception?